src/print.jl

```diff
@@ -27,6 +27,62 @@ abstract type PrintMode end
 abstract type REPLMode <: PrintMode end
 abstract type IJuliaMode <: PrintMode end
 
+struct _ModelPrinter{T<:AbstractModel}
+    model::T
+end
+
+function Base.show(io::IO, model::AbstractModel)
+    return _print_summary(io, model)
+end
+
+function Base.show(io::IO, ::MIME"text/plain", model::_ModelPrinter)
+    return _print_model(io, model.model)
+end
+
+function Base.show(io::IO, ::MIME"text/latex", model::_ModelPrinter)
+    return _print_latex(io, model.model)
+end
+
+"""
+    Base.print([io::IO = stdout,] model::AbstractModel; [latex::Bool])
+
+Print the formulation of `model` to `io` (which defaults to `stdout` if not
+provided).
+
+In notebooks and documentation, calling `print(model)` will render the LaTeX
+formulation to the screen.
+
+## Examples
+
+Print the formulation of the model to `stdout`, rendering as LaTeX if called
+from a notebook or the documentation, or plain-text if called from the REPL.
+
+    print(model)
+
+Print the plain-text formulation to `io` as a string (or to `stdout` if `io` is
+not given).
+
+    print(io, model; latex = false)
+    print(model; latex = false)
+
+Print the latex formulation to the screen as a string  (or to `stdout` if `io`
+is not given) .
+
+    print(io, model; latex = true)
+    print(model; latex = true)
+"""
+function Base.print(io::IO, model::AbstractModel; latex::Bool = false)
+    return latex ? _print_latex(io, model) : _print_model(io, model)
+end
+
+function Base.print(model::AbstractModel; latex::Union{Nothing,Bool} = nothing)
+    if latex === nothing
+        return _ModelPrinter(model)
```

Comment:
If `print(model)` is called from inside a function (so the output is not returned to the REPL or Jupyter), then this won't print anything right ? So shouldn't this be `show(_ModelPrinter(model))` ?

Reply:
Ah. Maybe `Base.print` is the wrong name for this. Calling `show` explicitly won't work. We need an object with `show` overloaded so that IJulia can pick the `text/latex` MIME type. If we call `show` directly, it will just print the plain-text.

Reply:
Maybe make it more explicit, e.g. the user should call `ModelPrinter(model)`.or `pretty_printing(model)`. `pretty_print` is missleading for the same reason `print` is.

Reply:
How's this now? I added `LaTeXify`.